client/my-sites/stats/hooks/use-intervals.tsx

```diff
@@ -21,52 +23,52 @@ type IntervalsType = {
 	[ key: string ]: IntervalType;
 };
 
-const intervals = {
-	[ STATS_PERIOD.DAY ]: {
-		id: STATS_PERIOD.DAY,
-		label: translate( 'Days' ),
-		statType: STATS_FEATURE_INTERVAL_DROPDOWN_DAY,
-	},
-	[ STATS_PERIOD.WEEK ]: {
-		id: STATS_PERIOD.WEEK,
-		label: translate( 'Weeks' ),
-		statType: STATS_FEATURE_INTERVAL_DROPDOWN_WEEK,
-	},
-	[ STATS_PERIOD.MONTH ]: {
-		id: STATS_PERIOD.MONTH,
-		label: translate( 'Months' ),
-		statType: STATS_FEATURE_INTERVAL_DROPDOWN_MONTH,
-	},
-	[ STATS_PERIOD.YEAR ]: {
-		id: STATS_PERIOD.YEAR,
-		label: translate( 'Years' ),
-		statType: STATS_FEATURE_INTERVAL_DROPDOWN_YEAR,
-	},
-};
+function useStaticIntervals() {
+	const translate = useTranslate();
+	return useMemo(
+		() => ( {
+			[ STATS_PERIOD.DAY ]: {
+				id: STATS_PERIOD.DAY,
+				label: translate( 'Days' ),
+				statType: STATS_FEATURE_INTERVAL_DROPDOWN_DAY,
+			},
+			[ STATS_PERIOD.WEEK ]: {
+				id: STATS_PERIOD.WEEK,
+				label: translate( 'Weeks' ),
+				statType: STATS_FEATURE_INTERVAL_DROPDOWN_WEEK,
+			},
+			[ STATS_PERIOD.MONTH ]: {
+				id: STATS_PERIOD.MONTH,
+				label: translate( 'Months' ),
+				statType: STATS_FEATURE_INTERVAL_DROPDOWN_MONTH,
+			},
+			[ STATS_PERIOD.YEAR ]: {
+				id: STATS_PERIOD.YEAR,
+				label: translate( 'Years' ),
+				statType: STATS_FEATURE_INTERVAL_DROPDOWN_YEAR,
+			},
+		} ),
+		[ translate ]
+	);
+}
 
-const getGatedIntervals = createSelector(
-	( state, siteId ) => {
-		return Object.keys( intervals ).reduce( ( acc, key ) => {
-			const interval = intervals[ key ];
+const getGatedIntervals = createSelector( ( state: AppState, siteId, intervals ) => {
+	return Object.keys( intervals ).reduce( ( acc, key ) => {
+		const interval = intervals[ key ];
 
-			return {
-				...acc,
-				[ key ]: {
-					...interval,
-					isGated: shouldGateStats( state, siteId, interval.statType ),
-				},
-			};
-		}, {} );
-	},
-	Object.values( intervals ).map(
-		( { statType } ) =>
-			( state: object, siteId ) =>
-				shouldGateStats( state, siteId, statType )
-	)
-);
+		return {
+			...acc,
+			[ key ]: {
+				...interval,
+				isGated: shouldGateStats( state, siteId, interval.statType ),
+			},
+		};
+	}, {} );
+} );
```

Comment:
Heya. I don't know if we should omit the `getDependents` param. The result is shallow-comparing the entire state object, which will clear the cache when any part of the state tree changes. Was that intentional?

Reply:
p.s. it might help to add a test for the hook to ensure memoization works as intended. This could have slipped through if unintended. I can help write the test when back.

Reply:
I've initially removed getDependents since having it resulted in 25% more calls (about 800) to shouldGateStats, so it seemed to be actually degrading performance. But if the idea is to avoid re-rendering, then it probably should be there, so I've added it back.
> p.s. it might help to add a test for the hook to ensure memoization works as intended. This could have slipped through if unintended. I can help write the test when back.

That'd be appreciated, my tests currently fail with some TS errors.